sd.py

```diff
@@ -54,16 +54,25 @@ def __init__(self, device, sd_version='2.1', hf_key=None):
             raise ValueError(f'Stable-diffusion version {self.sd_version} not supported.')
 
         # Create model
-        self.vae = AutoencoderKL.from_pretrained(model_key, subfolder="vae").to(self.device)
-        self.tokenizer = CLIPTokenizer.from_pretrained(model_key, subfolder="tokenizer")
-        self.text_encoder = CLIPTextModel.from_pretrained(model_key, subfolder="text_encoder").to(self.device)
-        self.unet = UNet2DConditionModel.from_pretrained(model_key, subfolder="unet").to(self.device)
+        if vram_O > 0:
+            pipe = StableDiffusionPipeline.from_pretrained(model_key, torch_dtype=torch.float16)
```

Comment:
consider using opt.fp16 to control data type? like passing in opt.fp16 in __init__ and use `torch_dtype=torch.float16 if fp16 else torch.float32`

Reply:
This looks nice. Will do it like this. Thanks